### PR TITLE
Add tx settings to give users more capabilities to customize events

### PR DESCRIPTION
**Pull request by nebolax**
Adds some settings that allow users to customize events.

These settings don't have counterparty set and will be applied when users customize events.

### CODE REVIEW

File: rotkehlchen/chain/ethereum/accounting/aggregator.py [hunk 1]
```diff
@@ -112,6 +113,52 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
...
+        airdrop_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.AIRDROP)
+        airdrops_taxable = LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions
+        result[airdrop_key] = TxEventSettings(
+            taxable=airdrops_taxable,
+            count_entire_amount_spend=False,
+            count_cost_basis_pnl=False,
+            method='acquisition',
+            take=1,
+            multitake_treatment=None,
+        )
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/accounting/aggregator.py:134]: This has to respect the setting for airdrops as we do in https://github.com/rotki/rotki/blob/296f830262c3022830d84130396549a445341fd1/rotkehlchen/chain/ethereum/modules/airdrops/accountant.py#L21

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+@pytest.fixture(name='pot')
+def fixture_pot(accountant):
+    pot: 'AccountingPot' = accountant.pots[0]
+    with pot.database.conn.read_ctx() as cursor:
+        settings = pot.database.get_settings(cursor)
+    pot.reset(
+        settings=settings,
+        start_ts=Timestamp(0),
+        end_ts=Timestamp(100),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:35]: how are this start/end chosen. Does it affect somehow the tests?

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        report_id=1,
+    )
+    return pot
+
+
+def gain_one_ether(transactions, evmhash):
+    eth_gain_event = HistoryBaseEntry(
+        event_identifier=evmhash,
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:43]: Can you add a docstring explaining what this function does and how it is used? Also if is only used here you can add the `_` before the name

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        location=Location.BLOCKCHAIN,
+        event_type=HistoryEventType.INFORMATIONAL,
+        event_subtype=HistoryEventSubType.NONE,
+        asset=A_ETH,
+        balance=Balance(),
+    )
+    was_skipped = False
+
+    def log_mock(text):
+        assert 'Skipping...' in text
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:75]: shouldn't you also check that the skipped entry is the one you think is being skipped? Doesn't it have the identifier in the row?

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+        nonlocal was_skipped
+        was_skipped = True
+
+    log_patch = patch('rotkehlchen.accounting.transactions.log.debug', side_effect=log_mock)
+    with log_patch:
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:80]: can it go in one line? if you don't need to reuse it I believe is fine to just do

```
with patch(...
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.history import prices
+from rotkehlchen.types import Location, Timestamp, TimestampMS, deserialize_evm_tx_hash
+
+
+@pytest.fixture(name='pot')
+def fixture_pot(accountant):
+    pot: 'AccountingPot' = accountant.pots[0]
+    with pot.database.conn.read_ctx() as cursor:
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:30]: this fixture should go with the other fixtures in the `tests/fixtures` folders. Probably under `accounting`. Also what do you think about renaming this one to `accounting_pot`?

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+        notes='Send 1 ETH to 0xABC',
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.NONE,
+    )
+    consumed_num = pot.transactions.process(event=gas_event, events_iterator=None)
+    assert consumed_num == 1
+
+    acquisition_event = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=Timestamp(1624395186),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:171]: This timestamp comes from the transaction in the method you have right? Maybe you can return it and use it instead of repeating this constant in all the tests

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+        extra_data={'tx_hash': tx_hex},
+    )
+    expected_event.calculate_pnl(count_entire_amount_spend=True, count_cost_basis_pnl=True)
+    assert pot.processed_events[-1] == expected_event
+
+
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_spend_settings(pot):
+    """Test that the default accounting settings for spends are correct"""
+    tx_hex = '0x282f632c7a2385d026068122dc415562a1c4e827fd54673671af1d40445d6b58'
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:149]: This txhash seems to be the same in all the tests. Can you set it as a constant and state that is not relevant for the test?

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+    pot.reset(
+        settings=settings,
+        start_ts=Timestamp(0),
+        end_ts=Timestamp(100),
+        report_id=1,
+    )
+    return pot
+
+
+def gain_one_ether(transactions, evmhash):
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:41]: Why do you need to pass the evmhash? is it used at some point?. The event is constant so it doesn't seem to be relevant

**Review by yabirgb — COMMENTED** [review]
Left some comments about the implementation.

About how to organize the tests for accounting I'm not sure. Since we already mock the transactions when preparing the tests for the decoders maybe is it easier to check that the accounting is properly done there? What do you think? Also @LefterisJP could also have another idea

File: rotkehlchen/chain/ethereum/accounting/aggregator.py (hunk 1 above)
Comment (nebolax) [reply rotkehlchen/chain/ethereum/accounting/aggregator.py:134]: Hmmm I don't get you, what setting?

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py (hunk 2 above)
Comment (nebolax) [reply rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:35]: Will add a comment into the code, but no, it does not matter, since is used only for querying history which does not happen in the tests in this file

File: rotkehlchen/chain/ethereum/accounting/aggregator.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/accounting/aggregator.py:134]: This one that sets if the airdrops are taxable https://github.com/rotki/rotki/blob/296f830262c3022830d84130396549a445341fd1/rotkehlchen/chain/ethereum/modules/airdrops/accountant.py#L19

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py (hunk 4 above)
Comment (nebolax) [reply rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:75]: No, it doesn't have the identifier. But in the place where this patch is used:

```python
with log_patch:
        consumed_num = accounting_pot.transactions.process(event=event, events_iterator=None)
```

only a single event is processed, so I think there is no chance that some other event will appear there

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py (hunk 5 above)
Comment (nebolax) [reply rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:80]: Alright. Wanted to be more consistent with the usages of patches in other tests, but I guess can also go in one line

File: rotkehlchen/chain/ethereum/accounting/aggregator.py (hunk 1 above)
Comment (nebolax) [reply rotkehlchen/chain/ethereum/accounting/aggregator.py:134]: Isn't this setting respected here? :thinking: 
image

Comment (yabirgb) [reply rotkehlchen/chain/ethereum/accounting/aggregator.py:134]: I would swear it was not there but yes, that one. Sorry

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+
+
+def _gain_one_ether(transactions):
+    """Helper function to gain 1 ETH, so that spending events have what to spend"""
+    eth_gain_event = HistoryBaseEntry(
+        event_identifier=EXAMPLE_EVM_HASH,
+        sequence_index=0,
+        timestamp=TIMESTAMP_1_MS,
+        location=Location.BLOCKCHAIN,
+        location_label=string_to_evm_address('0x4bBa290826C253BD854121346c370a9886d1bC26'),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:60]: this address is being used around all the tests. Can be a constant too

**Review by yabirgb — COMMENTED** [review]
This is also would need an update in the docs explaining this rules.

I was checking the tests and isn't this missing the tests for the rules that you created?

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 11]
```diff
@@ -155,6 +164,40 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
             take=1,
             method='acquisition',
         )
+        swap_key = str(HistoryEventType.TRADE) + '__' + str(HistoryEventSubType.SPEND)
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/accounting/aggregator.py:167]: why is this here? Don't all swaps already have their own rules in each accountant module? IF this is here then shouldn't those accountants be removed?

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 12]
```diff
@@ -123,6 +124,14 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
             take=1,
             method='spend',
         )
+        fee_key = str(HistoryEventType.SPEND) + '__' + str(HistoryEventSubType.FEE)
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/accounting/aggregator.py:127]: There is already `get_tx_event_type_identifier(HistoryEventType.SPEND, HistoryEventSubType.FEE, CPT_ONEINCH_V1): TxEventSettings(  # noqa: E501` for fees. Shoudl it exist? It has `count_entire_amount_spend` as `False` though.

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 13]
```diff
@@ -155,6 +164,40 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
...
+        swap_key = str(HistoryEventType.TRADE) + '__' + str(HistoryEventSubType.SPEND)
+        result[swap_key] = TxEventSettings(
+            taxable=True,
+            count_entire_amount_spend=False,
+            count_cost_basis_pnl=True,
+            method='spend',
+            take=2,
+            multitake_treatment=TxMultitakeTreatment.SWAP,
+        )
+        airdrop_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.AIRDROP)
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/accounting/aggregator.py:176]: why are you adding this here? We already have all the airdrop accountants set up and there should be no more airdrops existing outside of those.

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 14]
```diff
@@ -155,6 +164,40 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
...
+        airdrops_taxable = LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions
+        result[airdrop_key] = TxEventSettings(
+            taxable=airdrops_taxable,
+            count_entire_amount_spend=False,
+            count_cost_basis_pnl=False,
+            method='acquisition',
+            take=1,
+        )
+        reward_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.REWARD)
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/accounting/aggregator.py:185]: We already have this in aave1-v1, compound and votium. The only difference is aave-v1 has `count_cost_basis_pnl` as True.

The question here is why the discrepancy? Am I correct in seeing that **for acquisitions**, `count_cost_basis_pnl` is not used? So it's the same. So then again the question on why have them both in the accountants and here stays.

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 15]
```diff
@@ -155,6 +164,40 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
...
+        result[reward_key] = TxEventSettings(
+            taxable=True,
+            count_entire_amount_spend=False,
+            count_cost_basis_pnl=False,
+            method='acquisition',
+            take=1,
+        )
+        renew_key = str(HistoryEventType.RENEW) + '__' + str(HistoryEventSubType.NONE)
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/accounting/aggregator.py:193]: why make this a generic default? Why not just add it to the only place this exists, ens module as an accountant?

**Review by LefterisJP — COMMENTED** [review]
Okay sorry I don't get the point of this PR. (I did not check the tests though)

 You seem to confuse defaults with accountants and re-add the same things that should already exist and by doing so also just waste a lot of logic. If you wanted to add missing per-module definitions do that. Don't just add all in the defaults file. That's why we creates a modular system.

Defaults only existed for things that do not have a module, and thus can't have an accountant. You probably have not understood how this is supposed to work. Both you @yabirgb and @nebolax

**Comment by LefterisJP** [comment]
> Okay sorry I don't get the point of this PR. (I did not check the tests though)

Hmmm as I was thinking why you would try to do this yesterday I came up with a possible answer. Did you do all this in order to make it possible for swaps/airdrops etc. to be taken into account in a generic way when the user edits a transaction like that?

If yes ... then please guys write that somewhere in the description of the PR or in some comment. Don't say "Add some default tx settings".

Also in that case I would:

1. Add them in a separate function to separate what would be potentially editable combinations by the user from the defaults for which we don't build a module accountant.
2. Explain what they are in a docstring
3. (In the future) - they should also be part of configurability just like the accountant ones: https://github.com/rotki/rotki/issues/4341

File: rotkehlchen/chain/evm/accounting/aggregator.py (hunk 12 above)
Comment (nebolax) [reply rotkehlchen/chain/evm/accounting/aggregator.py:127]: Hm, I am not sure what value should `count_entire_amount_spend` have because I guess in different jurisdictions fees should be treated differently. I chose `True` here because gas by default is taxable and has `count_entire_amount_spend` set to `True`.

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 16]
```diff
@@ -157,6 +153,73 @@ def get_accounting_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSett
         )
         return result
 
+    def _get_settings_for_customization(self, pot: 'AccountingPot') -> dict[str, TxEventSettings]:
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/accounting/aggregator.py:156]: nitpick: Both in the name and in the docstring don't use customization as a word. Since that we should use for customizing the tax treatment of each different type of event when https://github.com/rotki/rotki/issues/4341 is done.

Perhaps call it editable_combinations and in the docstrings explain users can edit the transactions type/subtype to get them.

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 17]
```diff
@@ -0,0 +1,338 @@
+from typing import Optional
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:1]: Why not place this inside `tests/unit/accounting`? This is not a decoder test but an accounting test.

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 18]
```diff
@@ -0,0 +1,338 @@
...
+                taxable_ledger_actions=taxable_ledger_actions,
+                include_gas_costs=gas_taxable,
+            ),
+        )
+    with pot.database.conn.read_ctx() as cursor:
+        settings = pot.database.get_settings(cursor)
+    pot.reset(
+        settings=settings,
+        start_ts=Timestamp(0),  # timestamps don't matter since they are used only for querying
+        end_ts=Timestamp(0),  # histroy, which does not happen in the tests in this file
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:71]:
```suggestion
        end_ts=Timestamp(0),  # history, which does not happen in the tests in this file
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 19]
```diff
@@ -0,0 +1,338 @@
...
+    (HistoryEventType.RECEIVE, HistoryEventSubType.REWARD, True, False),
+])
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_receipts_settings(
+        accounting_pot: 'AccountingPot',
+        event_type: 'HistoryEventType',
+        event_subtype: 'HistoryEventSubType',
+        is_taxable: bool,
+):
+    """Test that the default accounting settings for recepts are correct"""
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:141]:
```suggestion
    """Test that the default accounting settings for receiving are correct"""
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 20]
```diff
@@ -0,0 +1,338 @@
...
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.db.settings import ModifiableDBSettings
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.history import prices
+from rotkehlchen.types import Location, Price, Timestamp, TimestampMS, deserialize_evm_tx_hash
+from rotkehlchen.utils.misc import ts_sec_to_ms
+
+
+# The transaction hash doesn't matter in the tests in this file, so we can use any.
+EXAMPLE_TX_HASH_HEX = '0x282f632c7a2385d026068122dc415562a1c4e827fd54673671af1d40445d6b58'
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:33]: In which case you should just use a random one and not even mention anything here.We have the `make_random_bytes(42)` which can easily be turned into a hexstring.

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 21]
```diff
@@ -0,0 +1,338 @@
...
+EXAMPLE_EVM_HASH = deserialize_evm_tx_hash(EXAMPLE_TX_HASH_HEX)
+EXAMPLE_ADDRESS = string_to_evm_address('0x4bBa290826C253BD854121346c370a9886d1bC26')
+
+# Some utility timestamps. They don't matter here and are used only for mocked prices.
+TIMESTAMP_1_SECS = Timestamp(1624395186)
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:38]: Again same thing. If no specific value is needed no need to write it in the comment. You show it by using `make_random_timestamp()`

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 22]
```diff
@@ -0,0 +1,338 @@
...
+    )
+    return pot
+
+
+def _gain_one_ether(
+        transactions: 'TransactionsAccountant',
+        event_type: 'HistoryEventType' = HistoryEventType.RECEIVE,
+        event_subtype: 'HistoryEventSubType' = HistoryEventSubType.NONE,
+) -> None:
+    """Helper function to gain 1 ETH, so that spending events have what to spend"""
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:82]:
```suggestion
    """Helper function to gain 1 ETH, so that spending events have something to spend"""
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 23]
```diff
@@ -0,0 +1,338 @@
...
+        event_subtype=HistoryEventSubType.NONE,
+        asset=A_ETH,
+        balance=Balance(),
+    )
+    was_skipped = False
+
+    def log_mock(text):
+        assert 'Skipping...' in text
+        nonlocal was_skipped
+        was_skipped = True
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:119]: Don't do this. Do not rely on a log entry like this in a test. Check what is not there. So no processed events, no pnl change etc.

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 24]
```diff
@@ -0,0 +1,338 @@
...
+        pnl=PNL(),
+        cost_basis=None,
+        index=0,
+        extra_data={'tx_hash': EXAMPLE_TX_HASH_HEX},
+    )
+    expected_event.calculate_pnl(
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=is_taxable,
+    )
+    assert accounting_pot.processed_events == [expected_event]
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:166]: shouldn't you also check that pnl is correclty calculated? That's the whole point of accounting.

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 25]
```diff
@@ -0,0 +1,338 @@
...
+@pytest.mark.parametrize('event_type,event_subtype,is_taxable,counterparty,gas_taxable', [
+    (HistoryEventType.SPEND, HistoryEventSubType.NONE, True, None, False),
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, True, None, False),
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, True, CPT_GAS, True),
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, False, CPT_GAS, False),
+    (HistoryEventType.DEPOSIT, HistoryEventSubType.NONE, False, None, False),
+    (HistoryEventType.RENEW, HistoryEventSubType.NONE, True, None, False),
+])
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_spents_settings(
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:178]:
```suggestion
def test_accounting_spend_settings(
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 26]
```diff
@@ -0,0 +1,338 @@
...
+        accounting_pot: 'AccountingPot',
+        event_type: 'HistoryEventType',
+        event_subtype: 'HistoryEventSubType',
+        is_taxable: bool,
+        counterparty: Optional[str],
+):
+    """Test that the default accounting settings for spends are correct"""
+    # Gain some ETH so that the spending event has something to spend
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:186]: name is already descriptive enough
```suggestion
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 27]
```diff
@@ -0,0 +1,338 @@
...
+        pnl=PNL(),
+        cost_basis=cost_basis,
+        index=1,
+        extra_data={'tx_hash': EXAMPLE_TX_HASH_HEX},
+    )
+    expected_event.calculate_pnl(
+        count_entire_amount_spend=is_taxable,
+        count_cost_basis_pnl=is_taxable,
+    )
+    assert accounting_pot.processed_events[-1] == expected_event
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:245]: again same thing. You don't seem to check the PnL at all. The entire point of accounting is to have correct PnL. This is what you should check

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 28]
```diff
@@ -0,0 +1,338 @@
...
+
+
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_swap_settings(accounting_pot: 'AccountingPot'):
+    """Test that the default accounting settings for deposits are correct"""
+    # Gain some ETH so that the spending event has something to spend
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:251]:
```suggestion
```

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py [hunk 29]
```diff
@@ -0,0 +1,338 @@
...
+        pnl=PNL(free=ZERO, taxable=ZERO),
+        cost_basis=None,
+        index=2,
+        extra_data={'tx_hash': EXAMPLE_TX_HASH_HEX, 'group_id': EXAMPLE_TX_HASH_HEX + '12'},
+    )
+    expected_receive_event.calculate_pnl(
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+    )
+    assert accounting_pot.processed_events[1:] == [expected_spend_event, expected_receive_event]
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:338]: same here for pnl

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py (hunk 17 above)
Comment (nebolax) [reply rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:1]: Yes, makes sense

File: rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py (hunk 21 above)
Comment (nebolax) [reply rotkehlchen/tests/unit/decoders/accounting_settings/test_default_settings.py:38]: Will adjust the comments, but timestamps are required to be preset because they are used for getting historical prices

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 30]
```diff
@@ -0,0 +1,332 @@
...
+    _gain_one_ether(transactions=accounting_pot.transactions)
+    spend_event = HistoryBaseEntry(
+        event_identifier=EXAMPLE_EVM_HASH,
+        sequence_index=0,
+        timestamp=TIMESTAMP_2_MS,
+        location=Location.BLOCKCHAIN,
+        location_label=EXAMPLE_ADDRESS,
+        asset=A_ETH,
+        balance=Balance(amount=FVal(0.5)),
+        notes='Send 1 ETH to 0xABC',
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:190]: You are sneind half an ETH, no?

```suggestion
        notes='Send 0.5 ETH to 0xABC',
```

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 31]
```diff
@@ -0,0 +1,332 @@
...
+                    amount=FVal(0.5),
+                    event=acquisition_event,
+                    taxable=True,
+                ),
+            ],
+            is_complete=True,
+        )
+    expected_event = ProcessedAccountingEvent(
+        type=AccountingEventType.TRANSACTION_EVENT,
+        notes='Send 1 ETH to 0xABC',
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:222]:
```suggestion
        notes='Send 0.5 ETH to 0xABC',
```

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 32]
```diff
@@ -0,0 +1,332 @@
...
+
+    acquisition_event = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=TIMESTAMP_1_SECS,
+        rate=Price(FVal(1862.06)),
+        index=0,
+    )
+    acquisition_event.remaining_amount = FVal(0.5)
+    cost_basis = None
+    if is_taxable:
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:206]: So some comments here:

1. Add comments, with the calculations and what it is you are testing.
2. You are mocking stuff. Use easier numbers. Why the heck use `1862.06` and `3258.55`. You know you can **and should** mock price queries. We have fixtures for this. Make sure no network calls happen in all these tests you made.
3. The spending part and the selling part are different and should be checked for in the test depending on customization. This is what `count_entire_amount` as spend is for and `taxable + crypto_to_crypto`.

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 33]
```diff
@@ -0,0 +1,332 @@
...
+    )
+    consumed_num = accounting_pot.transactions.process(
+        event=swap_spend_event,
+        events_iterator=iter([swap_receive_event]),
+    )
+    assert consumed_num == 2
+    acquisition_event = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=TIMESTAMP_1_SECS,
+        rate=Price(FVal(1862.06)),
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:282]: same comments for rates here. Also for numbers you use multiple times in a test, add them in a variable

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 34]
```diff
@@ -0,0 +1,332 @@
...
+    assert accounting_pot.processed_events[-1] == expected_event
+    assert accounting_pot.pnls.taxable == FVal(1862.06) + expected_event.pnl.taxable
+    assert accounting_pot.pnls.free == ZERO
+
+
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_swap_settings(accounting_pot: 'AccountingPot'):
+    """Test that the default accounting settings for swaps are correct"""
+    _gain_one_ether(transactions=accounting_pot.transactions)
+
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:249]:
```suggestion
```

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 35]
```diff
@@ -0,0 +1,332 @@
...
+    expected_receive_event = ProcessedAccountingEvent(
+        type=AccountingEventType.TRANSACTION_EVENT,
+        notes='Receive 1200 DAI as the result of a swap',
+        location=Location.BLOCKCHAIN,
+        timestamp=TIMESTAMP_2_SECS,
+        asset=A_DAI,
+        free_amount=FVal(1200),
+        taxable_amount=ZERO,
+        price=Price(FVal('2.715458333333333333333333333')),
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:322]: Why this weird number? 1 DAI ~= 1$. Why make this more complicated?

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 36]
```diff
@@ -0,0 +1,332 @@
...
+    assert consumed_num == 2
+    acquisition_event = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=TIMESTAMP_1_SECS,
+        rate=Price(FVal(1862.06)),
+        index=0,
+    )
+    acquisition_event.remaining_amount = ZERO
+
+    expected_spend_event = ProcessedAccountingEvent(
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/accounting/test_default_settings.py:287]: You need to specify what the prices at the time are and how the processed event prices are derived. And use easier numbers, but make it make sense. At the moment it's very very hard to understand what's happening here. I can't really say if the test is correct or not.

File: rotkehlchen/chain/evm/accounting/aggregator.py [hunk 37]
```diff
@@ -104,22 +105,17 @@ class EVMAccountingAggregators():
...
-            result.update(aggregator.get_accounting_settings(pot))
-
-        # Also add the default settings
         gas_key = str(HistoryEventType.SPEND) + '__' + str(HistoryEventSubType.FEE) + '__' + CPT_GAS  # noqa: E501
         result[gas_key] = TxEventSettings(
             taxable=pot.settings.include_gas_costs,
-            count_entire_amount_spend=True,
-            count_cost_basis_pnl=True,
+            count_entire_amount_spend=pot.settings.include_gas_costs,
+            count_cost_basis_pnl=pot.settings.include_gas_costs,
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/chain/evm/accounting/aggregator.py:118]: Yes, makes sense, thanks :+1:

File: rotkehlchen/tests/unit/accounting/test_default_settings.py [hunk 38]
```diff
@@ -0,0 +1,332 @@
...
+])
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_spend_settings(
+        accounting_pot: 'AccountingPot',
+        event_type: 'HistoryEventType',
+        event_subtype: 'HistoryEventSubType',
+        is_taxable: bool,
+        counterparty: Optional[str],
+):
+    # Gain some ETH so that the spending event has something to spend
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/tests/unit/accounting/test_default_settings.py:180]: Ouch, don't know

File: rotkehlchen/tests/unit/accounting/test_default_settings.py (hunk 32 above)
Comment (nebolax) [reply rotkehlchen/tests/unit/accounting/test_default_settings.py:206]: Used already existing big structure with mocked prices and because of that prices were so ugly. But now improved.

**Review by LefterisJP — APPROVED** [review]
Perfect. All green and tests look correct. Merging.
